source/configuration_manual/authentication/oauth2.rst

```diff
@@ -253,60 +267,91 @@ Oauth2 overrides some of the default HTTP client and SSL settings:
 You can override these and any other HTTP client or SSL settings by placing
 them inside :dovecot_core:ref:`oauth2` named filter.
 
-.. code-block:: none
 
-  ### OAuth2 password database configuration
+.. dovecot_core:setting:: oauth2_tokeninfo_url
+   :values: @string
+
+   URL for verifying token validity. Token is appended to the URL. Example:
+   ``oauth2_tokeninfo_url = http://endpoint/oauth/tokeninfo?access_token=``
+
+.. dovecot_core:setting:: oauth2_introspection_url
+   :values: auth, get, post, local
+
+   How introspection is made, valid values are
+
+   auth
+     GET request with Bearer authentication
+   get
+     GET request with token appended to URL
+   post
+     POST request with token=bearer_token as content
+   local
+     Attempt to locally validate and decode JWT token
+
+   There is no default value, you must explicitly choose one.
+
+.. dovecot_core:setting:: oauth2_force_introspection
+   :default: no
+   :values: @boolean
+
+   Force introspection even if tokeninfo contains wanted fields.
+   Set this to yes if you are using :dovecot_core:ref:`oauth2_active_attribute`.
+
+.. dovecot_core:setting:: oauth2_local_validation
+   :seealso: @dict
+   :values: @named_filter
+
+   A dictionary for fetching validation keys.
+
+   Example:
+
+   .. code-block:: none  
 
-  ## url for verifying token validity. Token is appended to the URL
-  # tokeninfo_url = http://endpoint/oauth/tokeninfo?access_token=
+     local_validation {
+       dict_driver = fs
+       fs posix {
+       }
+       fs_posix_prefix = /tmp/keys/
+     }
 
-  ## introspection endpoint, used to gather extra fields and other information.
-  # introspection_url = http://endpoint/oauth/me
+.. dovecot_core:setting:: oauth2_scope
+   :values: @boollist
 
-  ## How introspection is made, valid values are
-  ##   auth = GET request with Bearer authentication
-  ##   get  = GET request with token appended to URL
-  ##   post = POST request with token=bearer_token as content
-  ##   local = Attempt to locally validate and decode JWT token
-  # introspection_mode = auth
+   A space separated or boolean list of scopes of validity. 
```

Comment:
I think simply "A list of scopes of validity". Even if it's space-separated it's still a boolean list. And the setting type is pretty clearly visible, so I don't think it needs explicitly saying "boolean list".

Reply:
Also it can actually be comma or space separated. In any case, better to keep boollist's explanation in a single place that is just linked.